### PR TITLE
Add Makefile for development

### PR DESCRIPTION
**Pull request by cofri**
A Makefile is added to handle:

- dev environment creation
- testing (formatting / linting / unit tests)
- documentation generation

Moreover, the README is improved with logo and contributing section moved to CONTRIBUTING.md.

### CODE REVIEW

File: requirements_dev.txt [hunk 1]
```diff
@@ -0,0 +1,14 @@
+tox
```
Comment (thib-s) [inline requirements_dev.txt:1]: this seems to be redundant with requirements defined in setup.py

**Review by thib-s — COMMENTED** [review]
the requirement.txt are redundant with `setup.py` requirements, you can:
- option 1: move the requirements in `setup.py`
- option 2: load `requirements.txt` in `setup.py` 
The important point is to manage dependencies from the same spot

**Comment by cofri** [comment]
> the requirement.txt are redundant with `setup.py` requirements, you can:
> 
>     * option 1: move the requirements in `setup.py`
> 
>     * option 2: load `requirements.txt` in `setup.py`
>       The important point is to manage dependencies from the same spot

I've removed the requirements files. They were only used in the Makefile. The required packages are then directly installed using the setup.py file.

I've also bumped version to 1.3.0.

**Review by thib-s — APPROVED** [review]
LGTM

**Pull request merged**